src/ExtendedTableWidget.cpp

```diff
@@ -341,6 +347,17 @@ void ExtendedTableWidget::paste()
 
 }
 
+void ExtendedTableWidget::useAsFilter()
+{
+    QModelIndex index = selectionModel()->currentIndex();
+
+    // Abort if there's nothing to filter
+    if (!index.isValid() || !selectionModel()->hasSelection())
+      return;
+
+    m_tableHeader->setFilter(index.column(), "=" + model()->data(index).toString());
```

Comment:
One minor thing here: you should probably use the ```Qt::EditRole``` in the ```data()``` call here. We use that to get the actual, unmodified cell data instead of what is shown to the user. Without this, 1) a low 'Symbol limit in cell' option can mean that we're filtering for the truncated string, 2) filtering for a BLOB value would literally filter for 'BLOB' which doesn't seem right either, and 3) filtering for NULL values wouldn't work if the 'NULL fields Text' option is set to something other than 'NULL'.

When changing to ```Qt::EditRole``` you'll need to check for ```isNull()``` though and filter for '=NULL' if that's true. Currently this works sort of automatically but only as long as the null text is set to 'NULL' in the settings.